campaign-launcher/client/src/components/Header/index.tsx

```diff
@@ -105,66 +136,87 @@ const Header: FC = () => {
             {activeAddress && isConnected ? <Account /> : <ConnectWallet />}
           </Box>
 
-          <IconButton
-            edge="start"
-            sx={{ display: { xs: 'block', md: 'none' }, color: 'primary.main' }}
-            onClick={() => toggleDrawer(true)}
-          >
-            <MenuIcon />
-          </IconButton>
+          <Box display={{ xs: 'flex', md: 'none' }} gap={2} alignItems="center">
+            {activeAddress && isConnected ? <Account /> : <ConnectWallet />}
+            <IconButton
+              edge="start"
+              sx={{
+                display: { xs: 'flex', md: 'none' },
+                color: 'primary.main',
+              }}
+              onClick={toggleMenu}
+              ref={toggleButtonRef}
+            >
+              <MenuIcon />
+            </IconButton>
+          </Box>
+        </Toolbar>
 
-          <Drawer
-            anchor="right"
-            open={isDrawerOpen}
-            onClose={() => toggleDrawer(false)}
-            slotProps={{
-              paper: {
-                elevation: 0,
-                sx: { width: '75%', bgcolor: 'background.default' },
-              },
-            }}
-          >
-            <Box
+        <ClickAwayListener onClickAway={handleClickAway}>
+          <Collapse in={isMenuOpen} timeout="auto">
+            <Paper
+              elevation={10}
               sx={{
-                p: 2,
                 display: 'flex',
                 flexDirection: 'column',
-                height: '100%',
-                alignItems: 'center',
-                textAlign: 'center',
-                gap: 2,
-                fontSize: '18px',
-                fontWeight: 600,
+                bgcolor: 'background.default',
+                borderRadius: '0',
+                boxShadow: 'none',
+                p: 2,
+                pb: 4,
+                gap: 3,
               }}
             >
-              <IconButton
-                onClick={() => toggleDrawer(false)}
-                sx={{
-                  color: 'primary.main',
-                  position: 'absolute',
-                  top: 10,
-                  right: 10,
-                }}
-              >
-                <CloseIcon />
-              </IconButton>
-              <StyledLink to={ROUTES.SUPPORT} text="Support" />
-              <StyledLink to={ROUTES.DASHBOARD} text="Dashboard" />
+              <StyledLink
+                to={ROUTES.SUPPORT}
+                text="Support"
+                onClick={() => setIsMenuOpen(false)}
```

Comment:
Would it be more optimal to have it defined as function in `useCallback` and pass in all places where used?